src/rod/utils/gazebo.py

```diff
@@ -92,16 +103,19 @@ def process_model_description_with_sdformat(
                 fp.write(model_description_string)
                 fp.close()
 
-            cp = subprocess.run(
-                [str(gazebo_executable), "sdf", "-p", fp.name],
-                text=True,
-                stdout=subprocess.PIPE,
-                stderr=subprocess.STDOUT,
-            )
-
-        if cp.returncode != 0:
-            print(cp.stdout)
-            raise RuntimeError("Failed to process the input with sdformat")
+            try:
+                cp = subprocess.run(
+                    [str(gazebo_executable), "sdf", "-p", fp.name],
+                    text=True,
+                    stdout=subprocess.PIPE,
+                    stderr=subprocess.STDOUT,
+                )
+            except subprocess.CalledProcessError as e:
```

Comment:
This exception is raised only if `check=True`. Have you checked that the behavior is preserved after refactoring the code? If not, I'll ask for a new test this time because it's not the first occurrence that refactored code is not properly checked.

Reply:
Yes I missed that, thanks! Addressed in https://github.com/ami-iit/rod/pull/31/commits/9de543a31e5351dc51603c42a0c151fa6cda05f3